server/service/src/invoice/prescription/update/validate.rs

```diff
@@ -39,9 +37,16 @@ pub fn validate(
         check_patient_exists(connection, patient_id)?.ok_or(PatientDoesNotExist)?;
     }
 
-    if let Some(backdated_datetime) = &patch.backdated_datetime {
-        // Check that any lines already assigned won't create a ledger discrepancy
-        check_stock_available_at_date(connection, &invoice.id, backdated_datetime)?;
+    if patch.backdated_datetime.is_some() {
+        // Check if we have any lines allocated to this invoice, if so we can't backdate
+        let lines = InvoiceLineRepository::new(connection).query_by_filter(
+            InvoiceLineFilter::new().invoice_id(EqualFilter::equal_to(&patch.id)),
+        )?;
+        if lines.len() > 0 {
```

Comment:
there is count() method for InvoiceLineRepository

Reply:
Nice, good point will change...